hydrolib/core/io/friction/models.py

```diff
@@ -0,0 +1,145 @@
+import logging
+from pathlib import Path
+from typing import List, Literal, Optional, Set, Union
+
+from pydantic import Field
+from pydantic.class_validators import root_validator, validator
+
+from hydrolib.core.io.ini.models import INIBasedModel, INIGeneral, INIModel
+from hydrolib.core.io.ini.util import (
+    get_enum_validator,
+    get_from_subclass_defaults,
+    get_split_string_on_delimiter_validator,
+)
+from hydrolib.core.utils import str_is_empty_or_none
+
+from ..ini.util import make_list_validator
+
+logger = logging.getLogger(__name__)
+
+
```

Comment:
In the remarks section of [https://content.oss.deltares.nl/delft3d/manuals/D-Flow_FM_User_Manual_1D2D.pdf#section.C.15](https://content.oss.deltares.nl/delft3d/manuals/D-Flow_FM_User_Manual_1D2D.pdf#section.C.15) there are some rules, do we want to add validation for this, perhaps in another issue?